packages/tpetra/core/src/Tpetra_Details_crsUtils.hpp

```diff
@@ -0,0 +1,376 @@
+  // since we're adding a new feature upon which existing code does not rely,
+  // namely Export/Import to a StaticProfile CrsGraph.  However, watch out
+  // for fence()ing relating to UVM.
+  auto this_row_beg = row_ptr_beg(0);
+  auto this_row_end = row_ptr_end(0);
+  using range = Kokkos::pair<typename RowPtr::value_type, typename RowPtr::value_type>;
+  for (typename RowPtr::size_type i=0; i<num_row-1; i++) {
+
+    // First, copy over indices for this row
+    auto used_this_row = this_row_end - this_row_beg;
+    auto indices_old_subview =
+      subview(indices, range(this_row_beg, this_row_beg+used_this_row));
+    auto indices_new_subview =
+      subview(indices_new, range(row_ptr_beg(i), row_ptr_beg(i)+used_this_row));
+
+    // just call memcpy; it works fine on device if this becomes a kernel
+    using value_type = typename Indices::non_const_value_type;
+    memcpy(indices_new_subview.data(), indices_old_subview.data(),
+           used_this_row * sizeof(value_type));
+
+    // TODO could this actually have extra entries at the end of each row?
+    // If yes, then fill those entries with an "invalid" value.
+
+    // Before modifying the row_ptr arrays, save current beg, end for next iteration
+    this_row_beg = row_ptr_beg(i+1);
+    this_row_end = row_ptr_end(i+1);
+
+    auto used_next_row = row_ptr_end(i+1) - row_ptr_beg(i+1);
+    row_ptr_beg(i+1) = row_ptr_beg(i) + entries_this_row(i);
+    row_ptr_end(i+1) = row_ptr_beg(i+1) + used_next_row;
+  }
+  {
+    row_ptr_beg(num_row) = indices_new.size();
+    // Copy indices for last row
+    auto n = num_row - 1;
+    auto used_this_row = row_ptr_end(n) - row_ptr_beg(n);
+    auto indices_old_subview =
+      subview(indices, range(this_row_beg, this_row_beg+used_this_row));
+    auto indices_new_subview =
+      subview(indices_new, range(row_ptr_beg(n), row_ptr_beg(n)+used_this_row));
+    using value_type = typename Indices::non_const_value_type;
+    memcpy(indices_new_subview.data(), indices_old_subview.data(),
+           used_this_row * sizeof(value_type));
+  }
+  indices = indices_new;
+}
+
+
+/// \brief Merges two ranges [first1, last1) and [first2, last2) into one sorted
+///   range beginning at d_first, eliminates all but the first element from
+///   every consecutive group of equivalent elements from the merged range,
+///   and returns a past-the-end iterator for the new logical end of the range.
+///
+///   On input, the first range need not be sorted, but the second must be
+///   sorted. The intended use case is the first range contains new elements to
+///   merge in to the existing CRS indices contained in the second range - which
+///   should already be sorted.
+///
+/// \param first1, last1 - first range of elements to merge (need not be sorted)
+/// \param first2, last2 - second range of elements to merge (must be sorted)
+///
+/// \return last - Output iterator to element past the last unique element in
+///   the merged range.
+///
```

Comment:
@mhoemmen, this new version of `ind_merge` uses `inplace_merge`.  This version seems much cleaner than before.